client/components/close-on-escape/index.jsx

```diff
@@ -0,0 +1,71 @@
+/**
+ * External dependencies
+ */
+import { filter, includes, isEmpty, last, noop } from 'lodash';
+import PropTypes from 'prop-types';
+import { Component } from 'react';
+
+const ESC_KEY_CODE = 27;
+let components = [];
+
+function onKeydown( event ) {
+	if (
+		! isEmpty( components ) &&
+		event.keyCode === ESC_KEY_CODE &&
+		! isInput( event.target )
+	) {
+		const component = last( components );
+
+		component.onEscape();
+	}
+}
+
+function isInput( element ) {
+	return includes( [ 'INPUT', 'TEXTAREA' ], element.nodeName );
```

Comment:
Turns out that we're passing the _close method to the `react-modal` component as it's `onRequestClose` prop.
I'll adjust as necessary :)